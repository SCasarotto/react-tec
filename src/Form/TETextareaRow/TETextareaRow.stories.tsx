import React from 'react'
import { Story, Meta } from '@storybook/react'

import { TETextareaRow, TETextareaRowProps } from './TETextareaRow'

export default {
	title: 'Form/TETextareaRow',
	component: TETextareaRow,
	argTypes: {},
} as Meta

const Template: Story<TETextareaRowProps> = (args) => <TETextareaRow {...args} />

export const Default = Template.bind({})
Default.args = {
	title: 'My Text Area',
	labelForKey: 'SomeUniqueKey42',
	onChange: (e) => console.log(e.target.value),
}
