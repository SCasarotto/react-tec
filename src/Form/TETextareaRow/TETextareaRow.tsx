import React from 'react'

import { TERow, TERowCustomProps } from '../TERow'
import { TELabel, TELabelCustomProps } from '../TELabel'
import { TETextarea, TETextareaProps } from '../TETextarea'

export interface TETextareaRowProps extends TERowCustomProps, TELabelCustomProps, TETextareaProps {
	labelForKey: string
}
/**
 *
 * A component containing a `TERow`, a `TELabel` and either a `TETextarea`. In addition to the props below, reference all props for `TETextarea`.
 *
 */
export const TETextareaRow = React.forwardRef<HTMLTextAreaElement, TETextareaRowProps>(
	(props, ref) => {
		const {
			rowSize,
			last,
			className = '',
			title,
			onChange,
			disabled,
			value,
			required,
			labelForKey,
			...rest
		} = props

		return (
			<TERow rowSize={rowSize} last={last} className={`TEInputRow ${className}`}>
				{title && (
					<TELabel
						htmlFor={labelForKey}
						required={required}
						disabled={disabled}
						className='TEInputRowTitle'
					>
						{title}
					</TELabel>
				)}
				<TETextarea
					value={value}
					onChange={onChange}
					id={labelForKey}
					disabled={disabled}
					className='TEInputRowInput'
					ref={ref}
					{...rest}
				/>
			</TERow>
		)
	},
)
