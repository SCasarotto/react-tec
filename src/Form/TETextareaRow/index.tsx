import React from 'react'

import { TERow, TERowCustomProps } from '../TERow'
import { TELabel, TELabelCustomProps } from '../TELabel'
import { TETextarea, TETextareaProps } from '../TETextarea'

export interface TETextareaRowProps
	extends TERowCustomProps,
		TELabelCustomProps,
		TETextareaProps {
	labelForKey: string
}

export const TETextareaRow: React.FC<TETextareaRowProps> = (props) => {
	const {
		rowSize,
		last,
		className = '',
		title,
		onChange,
		disabled,
		value,
		required,
		labelForKey,
		...rest
	} = props

	return (
		<TERow
			rowSize={rowSize}
			last={last}
			className={`TEInputRow ${className}`}
		>
			{title && (
				<TELabel
					htmlFor={labelForKey}
					required={required}
					disabled={disabled}
					className="TEInputRowTitle"
				>
					{title}
				</TELabel>
			)}
			<TETextarea
				value={value}
				onChange={onChange}
				id={labelForKey}
				disabled={disabled}
				className="TEInputRowInput"
				{...rest}
			/>
		</TERow>
	)
}
