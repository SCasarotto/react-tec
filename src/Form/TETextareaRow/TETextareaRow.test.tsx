import React from 'react'
import { render } from '@testing-library/react'
import { TETextareaRow } from './TETextareaRow'

describe('TETextareaRow', () => {
	// Renders Without Error
	it('matches snapshot', () => {
		const { asFragment } = render(<TETextareaRow labelForKey='testTETextareaRow' />)
		expect(asFragment()).toMatchSnapshot()
	})
})
