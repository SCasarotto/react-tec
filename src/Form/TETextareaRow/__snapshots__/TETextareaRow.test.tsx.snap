// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TETextareaRow matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnylx bbvaGu TERow TEInputRow "
  >
    <textarea
      class="sc-dlnjPT hsCBaU TETextarea TEInputRowInput"
      id="testTETextareaRow"
    />
  </div>
</DocumentFragment>
`;
